my_each.rb

```diff
@@ -1,6 +1,168 @@
 module Enumerable
-    def my_each(array)
-      for i in 0...array.length
-        p "gaa"
+  def my_each
+    array = to_a
+    (0...array.size).each do |i|
+      return to_enum(:my_each) unless block_given?
+
+      yield array[i]
     end
-end
\ No newline at end of file
+    self
+  end
+
+  def my_each_with_index
+    array = to_a
+    (0...array.size).each do |i|
+      return to_enum(:my_each_with_index) unless block_given?
+
+      yield array[i], i
+    end
+    self
+  end
+
+  def my_select
+    new_array = []
+    my_each do |element|
+      return to_enum(:my_select) unless block_given?
+
+      new_array.push(element) if yield element
+    end
+    new_array
+  end
+
+  def my_all?(compare = nil)
+    result = true
+    my_each do |element|
+      if compare
+        result = compare.class.eql?(Class) ? element.is_a?(compare) : element.eql?(compare)
+      elsif block_given?
+        result = yield element
+
+      end
+      break unless result
+    end
+
+    result
+  end
+
+  def my_any?(compare = nil)
+    result = true
+    my_each do |element|
+      if compare
+        result = compare.class.eql?(Class) ? element.is_a?(compare) : element.eql?(compare)
+      elsif block_given?
+        result = yield element
+      end
+      break if result
+    end
+    result
+  end
+
+  def my_none?(compare = nil)
+    result = true
+    my_each do |element|
+      result = if compare
+                 compare.class.eql?(Class) ? !element.is_a?(compare) : !element.eql?(compare)
+               elsif block_given?
+                 !(yield element)
+               else
+                 !element
+               end
+      break unless result
+    end
+    result
+  end
+
+  def my_count(arg)
+    counter = 0
+    param = element
+    my_each do |element|
+      param = if arg
+                arg
+              elsif block_given?
+                yield element
+              end
+      counter += 1 if param.eql?(true) || param.eql?(element)
+    end
+    counter
+  end
+
+  def my_map(proc = nil)
+    new_arr = []
+    my_each do |element|
+      param = if proc
+                proc.call(element)
+              else
+                return to_enum(:my_map) unless block_given?
+
+                yield element
+              end
+      new_arr.push(param)
+    end
+    new_arr
+  end
+
+  def my_inject(initial_value = 1, use = :+)
+    if initial_value.is_a?(Symbol)
+      use = initial_value
+      initial_value = 1
+    end
+    initial_value = 0 if use == :+
+
+    my_each do |element|
+      result = use&.to_proc&.call(initial_value, element)
+      result = yield initial_value, element if block_given?
+      initial_value = result
+    end
+    initial_value
+  end
+end
+
+def multiply_els
+  my_inject(1) do |acumulator, element|
+    acumulator * element
+  end
+end
```

Comment:
multiply_els should accept a parameter. I can not see it. The methods should look something like
```ruby
def multiply_els array
  array.my_inject(1) do |acumulator, element|
    acumulator * element
  end
end